docs/iris/src/whatsnew/make.py

```diff
@@ -0,0 +1,120 @@
+## (C) British Crown Copyright 2015, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris. If not, see <http://www.gnu.org/licenses/>.
+
+import datetime
+import os
+import sys
+import re
+from operator import itemgetter
+from distutils import version
+
+SOFTWARE_NAME = "Iris"
+VALID_CATEGORIES = [
+    {"Prefix": "bugfix", "Title": "Bugs Fixed"},
+    {"Prefix": "newfeature", "Title": "Features"},
+    {"Prefix": "incompatiblechange", "Title": "Incompatible Changes"},
+    {"Prefix": "deprecate", "Title": "Deprecations"},
+    {"Prefix": "docchange", "Title": "Documentation Changes"}
+]
+
+def get_release_number(directory_name):
+    """ Takes a formatted string and returns a LooseVersion object. """
+    regular_expression = "(.*)\\_(.*)"
+    compiled_filter = re.compile(regular_expression)
+    compilable_files_unsorted = []
+    directory_elements = compiled_filter.findall(directory_name)
+    try:
+        return version.LooseVersion(directory_elements[0][1])
+    except IndexError:
+        return None
+
+def get_directory_list(root_directory):
+    """ Returns a list of subdirectories, excluding files. """
+    directory_contents = os.listdir(root_directory)
+    subdirectories = []
+    for inode in directory_contents:
+        if os.path.isdir(inode):
+            subdirectories.append(inode)
+    return subdirectories
+
+def check_compiled(directory_name):
+    """ Checks if a release.rst file exists for each contribution directory. """
+    extension = ".rst"
+    release = get_release_number(directory_name)
+    compiled_filename = str(release) + extension
+    try:
```

Comment:
i'm puzzled why you need a try|except block here, I would prefer to see a test which returns true/false
are you jjust checking for the presence of a file?
what is the path that leads you to trap OSError instances?